descqa/CheckAstroPhoto.py

```diff
@@ -0,0 +1,139 @@
+
+        axHistx.autoscale(tight=True)
+        axHisty.autoscale(tight=True)
+        axHistx.set_xlim(axScatter.get_xlim())
+        axHisty.set_ylim(axScatter.get_ylim())
+        plt.savefig(savename)
+
+    def conclude_test(self, output_dir):
+        """
+        This function should gather the two catalogs, match them and perform the summary plots
+        """
+
+        try:
+            assert(len(self.ra.keys())==2) # Making sure that we have *just* two catalogs
+        except AssertionError:
+            print('The test can compare two catalogs only!')
+
+        cat_names = list(self.ra.keys()) # This is an auxiliary list to easily get the catalogs
+        print(cat_names)
+        cat_len = [len(self.ra[cat_names[0]]),len(self.ra[cat_names[1]])]
+        cat_names = np.array(cat_names)[np.argsort(cat_len)].tolist() # We find the catalog with less objects to build the tree
+        # For this test we are going to match using closest neighbor since it is the fastest but it can be easily
+        # swapped for any other matching strategy
+        
+        matched_id = spatial_closest(self.ra[cat_names[0]], self.dec[cat_names[0]], 
+                                     self.ra[cat_names[1]], self.dec[cat_names[1]], np.arange(cat_len[1]))[1]
+        
+        delta_ra = self.ra[cat_names[0]]-self.ra[cat_names[1]][matched_id]
+        delta_dec = self.dec[cat_names[0]]-self.dec[cat_names[1]][matched_id]
+        delta_mag = dict()
+        good_mag = dict()
+        for band in self.bands:
+            delta_mag[band] = self.magnitude[(cat_names[0], band)]-self.magnitude[(cat_names[1],band)][matched_id]
+            good_mag[band] = (np.isnan(self.magnitude[(cat_names[0], band)])==False) & (np.isinf(self.magnitude[(cat_names[0], band)])==False)
+        # Scatter plot + histogram of RA and Dec (assumed to be in degrees)
+        astro_savename = os.path.join(output_dir, 'astrometry_check_%s_%s.png' % (cat_names[0], cat_names[1]))
+        self.scatter_project(delta_ra*3600, delta_dec*3600, -0.5, 0.5, -0.5, 0.5, 100, r'$\Delta$ RA [arcsec]',
+                            r'$\Delta$ Dec [arcsec]', astro_savename)
+        
+        # Scatter plot + histogram of Delta mag vs mag
+        for band in self.bands:
+            photo_savename = os.path.join(output_dir, 'photometry_check_%s_%s_%s.png' % (cat_names[0], cat_names[1], band))
+            self.scatter_project(self.magnitude[(cat_names[0], band)][good_mag[band]],
+                delta_mag[band][good_mag[band]], self.min_mag, self.max_mag,
+                -1, 1, self.nbins, '%s' % band, r'$\Delta %s$' % band, photo_savename, bin_stat=True)
+            n_true, _ = np.histogram(self.magnitude[(cat_names[1], band)], bins=50, range=(10, 30))
+            n_meas, bin_edges = np.histogram(self.magnitude[(cat_names[0], band)], bins=50, range=(10, 30))
+            plt.figure()
+            plt.plot(0.5*(bin_edges[1:]+bin_edges[:-1]), 1.0*n_meas/n_true,'o')
```

Comment:
Instead of `1.0*n_meas/n_true`, do `n_meas.astype(float) / n_true` to be explicit what the code does.